api/librdb-api.h

```diff
@@ -193,15 +200,22 @@ typedef struct RdbHandlersRawCallbacks {
 typedef struct RdbHandlersStructCallbacks {
     HANDLERS_COMMON_CALLBACKS
     RdbRes (*handleStringValue)(RdbParser *p, void *userData, RdbBulk str);
-    RdbRes (*handleListLP)(RdbParser *p, void *userData, RdbBulk listLP);
+    /* list */
+    RdbRes (*handleListPlain)(RdbParser *p, void *userData, RdbBulk node);
```

Comment:
i don't think "legacy" is right. it should describe the format, not it's history.
ZL is also legacy, and LP might be some day too.
does this callback handle only the old RDB_TYPE_LIST? or also the QUICKLIST_NODE_CONTAINER_PLAIN item in RDB_TYPE_LIST_QUICKLIST_2?

Reply:
It also handles QUICKLIST_NODE_CONTAINER_PLAIN. Maybe we better distinct between the two?

Reply:
AFAIK, the only two use cases for this struct callbacks are:
1. memory usage estimation 
2. populating a db in redis itself

i suppose that either of them, can make all the necessary decision on what to do with the info, even if it uses the same callback (without a need to make a distinction between them).
on the other hand, the higher level set of callbacks is suppose to give you the data when you care about the data, and this lower level set of callbacks are suppose to give you raw data on how it is encoded in the rdb.

bottom line, i don't have a clear opinion on which is better, but only after we decide if we wanna split them we should choose names.
i suppose that if we split them, then it'll be [handleListZL, handleListLP, handleListElementSingle, handleListElementPlain]
if we keep them together, it should be [handleListZL, handleListLP, handleListElement]
WDYT?

Reply:
I prefer to have it simple for now:  [handleListZL, handleListLP, handleListElement].
Note that I escaped from using the term `handle` since it is already used by application handlers. I used instead `element`. Please consider my note below about it.

Reply:
so "element" is taken and we can't use handleListElement either.
in that case we can use "item" for list, "field", and "member" for set.
that should be reflected in RdbHandlersDataCallbacks too.

i don't mind not using the term "handle", but we must define what's an "element" and if it can also refer to a complete key, we can't use it to indicate a field.
IMHO, i'd rather avoid using it as you do now and reserve it for sub-key items only, but i'm ok with your way as long as we don't also use it for fields.